plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/impl/PluginManager.java

```diff
@@ -265,29 +472,46 @@ public JSONArray getPluginDependencyJsonArray(Plugin plugin, JSONObject ucJson)
      *
      * @param plugin
      */
-    public void resolveDependenciesFromManifest(Plugin plugin) {
-        String dependencyString = getAttributefromManifest(plugin.getFile(), "Plugin-Dependencies");
-        String[] dependencies = dependencyString.split(",");
+    public List<Plugin> resolveDependenciesFromManifest(Plugin plugin) {
         List<Plugin> dependentPlugins = new ArrayList<>();
 
-        System.out.println(plugin.getName() + " depends on: ");
-
-        boolean isPluginOptional = false;
+        try {
+            File tempFile = Files.createTempFile(plugin.getName(), ".jpi").toFile();
+            if (!downloadPlugin(plugin, tempFile)) {
+                System.out.println("Unable to resolve dependencies");
+                Files.delete(tempFile.toPath());
+                return dependentPlugins;
+            }
+            String dependencyString = getAttributefromManifest(tempFile, "Plugin-Dependencies");
+            String[] dependencies = dependencyString.split(",");
 
-        for (String dependency : dependencies) {
-            if (dependency.contains("resolution:=optional")) {
-                dependency = dependency.split(";")[0];
-                isPluginOptional = true;
+            if (isVerbose) {
+                System.out.println("\n" + plugin.getName() + " depends on: ");
```

Comment:
Maybe incomplete logging statement?

Reply:
The dependencies are output below. I could just print them all together at the end though and maybe that would be more clear.